basiclti/basiclti-portlet/src/java/org/sakaiproject/portlets/IMSBLTIPortlet.java

```diff
@@ -383,7 +383,7 @@ public void prepareEdit(RenderRequest request)
 				for(ToolConfiguration tool : (List<ToolConfiguration>) page.getTools()) {
 					String tid = tool.getToolId();
 					if ( "sakai.lessonbuildertool".equals(tid) ) foundLessons = true;
-					if ( "sakai.gradebook.tool".equals(tid) || "sakai.gradebook.gwt.rpc".equals(tid) ) foundGradebook = true;
+					if ( tid != null && tid.startsWith("sakai.gradebook") ) foundGradebook = true;
```

Comment:
This should really be checking a list or array of toolids so that tools aren't forced into being `sakai.x`
Alternatively, `StringUtils.startsWith` to remove the double check.